DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public class AnonymizerRuleFactory : IAnonymizerRuleFactory
+    {
+        private readonly AnonymizerDefaultSettings _defaultSettings;
+
+        private readonly Dictionary<string, JObject> _customizedSettings;
+
+        private readonly IAnonymizerProcessorFactory _processorFactory;
+
+        private readonly IAnonymizerSettingsFactory _settingsFactory;
+
+        public AnonymizerRuleFactory(AnonymizerConfiguration configuration, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _defaultSettings = configuration.DefaultSettings;
+            _customizedSettings = configuration.CustomizedSettings;
+            _processorFactory = processorFactory;
+            _settingsFactory = settingsFactory;
+        }
+
+        public AnonymizerRule[] CreateAnonymizationDicomRule(JObject[] ruleContents)
+        {
+            return ruleContents?.Select(entry => CreateAnonymizationDicomRule(entry)).ToArray();
+        }
+
+        public AnonymizerRule CreateAnonymizationDicomRule(JObject ruleContent)
+        {
+            EnsureArg.IsNotNull(ruleContent, nameof(ruleContent));
+
+            // Parse and validate method
+            if (!ruleContent.ContainsKey(Constants.MethodKey))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingConfigurationFields, "Missing method in rule config");
+            }
+
+            var method = ruleContent[Constants.MethodKey].ToString();
+            var supportedMethods = Enum.GetNames(typeof(AnonymizerMethod)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+            if (!supportedMethods.Contains(method))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationRule, $"Anonymization method {method} not supported.");
+            }
+
+            // Parse and validate settings
+            JObject parameters = null;
+            if (ruleContent.ContainsKey(Constants.Parameters))
+            {
+                parameters = ruleContent[Constants.Parameters].ToObject<JObject>();
+            }
+
+            JObject ruleSetting = _defaultSettings.GetDefaultSetting(method);
```

Comment:
>ruleSetting [](http://example.com/codeflow?start=20&length=11)

FYI. I don't go through the setting logic here carefully, so please use UTs to make your code handle all kinds of situations well.

Reply:
I will add more tests later.